src/kernel.ts

```diff
@@ -157,7 +151,7 @@ interface IKernelFuture extends IDisposable {
  * used to communicate with the kernel.
  */
 export
-class Kernel {
+class Kernel extends utils.ReadyPromise {
```

Comment:
Instead of inheriting from `ReadPromise`, make `ReadPromise` an object which is composed by the consumer. That will allow you to make the protected methods of `ReadyPromise` public, without cluttering up the API of the `Kernel` object. `Kernel` **has a** `ReadyPromise`, not *is a* `ReadyPromise`.